Utils/tests/Expect/iterableElementRuntimeIs_A01_t01.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion static void iterableElementsRuntimeIs<T>(dynamic it)
+/// Failure if `it` is not instance of `Iterable<T>` or has an element type of
+/// which is not exactly the `T`
+///
+/// @description Checks no exception is thrown when `it` is an `Iterable<T>` and
+/// all elements of it have a type `T` (not a subtype)
+/// @author sgrekhov22@gmail.com
+
+import "dart:async";
+import "../../../Utils/expect.dart";
+
+main() {
+  Expect.iterableElementsRuntimeIs<int>([1, 2, 3]);
```

Comment:
A similar comment as for Utils/tests/Expect/iterableElementRuntimeIsNot_A01_t01.dart would fit here.